src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -18,16 +18,25 @@ internal partial class SwitchBinder : LocalScopeBinder
         private TypeSymbol _switchGoverningType;
         private readonly GeneratedLabelSymbol _breakLabel;
 
-        private bool PatternsEnabled =>
-            (_switchSyntax?.SyntaxTree?.Options as CSharpParseOptions)?.IsFeatureEnabled(MessageID.IDS_FeaturePatternMatching) == true;
-
-        internal SwitchBinder(Binder next, SwitchStatementSyntax switchSyntax)
+        protected SwitchBinder(Binder next, SwitchStatementSyntax switchSyntax)
             : base(next)
         {
             _switchSyntax = switchSyntax;
             _breakLabel = new GeneratedLabelSymbol("break");
         }
 
+        internal static SwitchBinder Create(Binder next, SwitchStatementSyntax switchSyntax)
+        {
+            // When pattern-matching is enabled, we use a completely different binder and binding
+            // strategy for switch statements. Once we have confirmed that it is totally upward
+            // compatible with the existing syntax and semantics, we will remove *this* binder
+            // and use the new one for binding all switch statements.
```

Comment:
I am also curious what is the plan to get the confirmation? 

---
In reply to: [64946958](https://github.com/dotnet/roslyn/pull/11485#discussion_r64946958) [](ancestors = 64946958)